code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -134,76 +134,88 @@
-						return
+	if(exchange_parts(user, I))
+		return
 
-				if(panel_open && istype(I, /obj/item/crowbar))
-						default_deconstruction_crowbar(I)
-						return
+	if(anchored && !beaker)
+		if(default_deconstruction_screwdriver(user, "juicer_open", "juicer0", I))
+			return
 
-		if (istype(I, /obj/item/reagent_containers) && (I.container_type & OPENCONTAINER) )
-				if(beaker)
-						to_chat(user, "<span class='warning'>There's already a container inside.</span>")
-				else if(panel_open)
-						to_chat(user, "<span class='warning'>Close the maintenance panel first.</span>")
-				else
-						if(!user.drop_item())
-								return 1
-						beaker =  I
-						beaker.loc = src
-						update_icon()
-						src.updateUsrDialog()
-				return 1 //no afterattack
-
-		if(is_type_in_list(I, dried_items))
-				if(istype(I, /obj/item/reagent_containers/food/snacks/grown))
-						var/obj/item/reagent_containers/food/snacks/grown/G = I
-						if(!G.dry)
-								to_chat(user, "<span class='warning'>You must dry that first!</span>")
-								return 1
-
-		if(holdingitems && holdingitems.len >= limit)
-				to_chat(usr, "The machine cannot hold anymore items.")
-				return 1
-
-		//Fill machine with a bag!
-		if(istype(I, /obj/item/storage/bag))
-				var/obj/item/storage/bag/B = I
-				for (var/obj/item/reagent_containers/food/snacks/grown/G in B.contents)
-						B.remove_from_storage(G, src)
-						holdingitems += G
-						if(holdingitems && holdingitems.len >= limit) //Sanity checking so the blender doesn't overfill
-								to_chat(user, "<span class='notice'>You fill the All-In-One grinder to the brim.</span>")
-								break
+		if(panel_open && istype(I, /obj/item/crowbar))
+			default_deconstruction_crowbar(I)
+			return
 
-				if(!I.contents.len)
-						to_chat(user, "<span class='notice'>You empty the plant bag into the All-In-One grinder.</span>")
+	if(istype(I, /obj/item/reagent_containers) && (I.container_type & OPENCONTAINER) )
+		if(beaker)
+			to_chat(user, "<span class='warning'>There's already a container inside.</span>")
+		else if(panel_open)
+			to_chat(user, "<span class='warning'>Close the maintenance panel first.</span>")
+		else
+			if(!user.drop_item())
+				return TRUE
+			beaker =  I
+			beaker.loc = src
+			update_icon()
+			updateUsrDialog()
+		return TRUE //no afterattack
+
+	if(is_type_in_list(I, dried_items))
+		if(istype(I, /obj/item/reagent_containers/food/snacks/grown))
+			var/obj/item/reagent_containers/food/snacks/grown/G = I
+			if(!G.dry)
+				to_chat(user, "<span class='warning'>You must dry that first!</span>")
+				return TRUE
+
+	if(holdingitems && holdingitems.len >= limit)
+		to_chat(usr, "The machine cannot hold anymore items.")
+		return TRUE
+
+	//Fill machine with a bag!
+	if(istype(I, /obj/item/storage/bag))
+		var/obj/item/storage/bag/B = I
+		if(!B.contents.len)
+			to_chat(user, "<span class='warning'>[B] is empty.</span>")
+			return FALSE
```

Comment:
I think the logic is still inverted here (and below)?
In the rest of the proc, `TRUE` seems to be returned on failure of any kind, and `FALSE` on success; but it's the opposite in bag handling.
(Honestly I'm not sure `TRUE`/`FALSE` make it any better than `0`/`1` at all... but that's tangential)

Reply:
You're right, the return logic is reverse for a lot of the other stuff. It should be `TRUE` on success and `FALSE` on failure. I'll have to change it. I've been told to return T/F instead of 1/0, so that's why I'm doing that